pkg/packetfilter/nftables/namedset.go

```diff
@@ -63,20 +64,18 @@ func (n *namedSet) Name() string {
 
 func (n *namedSet) Flush() error {
 	tx := n.nftables.NewTransaction()
-
 	tx.Flush(&n.set)
+
 	err := n.nftables.Run(context.TODO(), tx)
+	if knftables.IsNotFound(err) {
+		return nil
+	}
 
 	return errors.Wrapf(err, "error flushing set %q", n.Name())
 }
 
 func (n *namedSet) Destroy() error {
-	tx := n.nftables.NewTransaction()
-	tx.Delete(&n.set)
-
-	err := n.nftables.Run(context.TODO(), tx)
-
-	return errors.Wrapf(err, "error deleting set %q", n.Name())
+	return deleteSet(n.nftables, n.set.Name)
```

Comment:
Maybe we should be consistent and use one of  'n.set.Name' or 'n.Name()' in this file

Reply:
Changed all occurrences to `n.Name()`.